build: Use ad-hoc text file to carry build arguments

ddn0 opened the pull request:
The ability of CMake to escape quotes in generator variables is
rudimentary. So instead of writing well-formed JSON, output simple
key=value lines.

ddn0 commented on .github/workflows/cpp.yaml:
```diff
@@ -191,10 +191,13 @@ jobs:
           Release)
             BUILD_TYPE=Release
-            NUM_TEST_THREADS=8
+            NUM_TEST_THREADS=2
```
setting `NUM_TEST_THREADS` was a hack to test a varying number of partitions in the `lonestar/CMakeLists.txt` tests: https://github.com/KatanaGraph/katana/blob/master/lonestar/CMakeLists.txt#L237

Can we separate the idea the `NUM_TEST_THREADS` from the number of partitions tested? Preferably in a way that doesn't introduce more configure time variables?

l-hoang replied:
Let me think about it. The problem with this change as you said is that we have no more 2+ partition tests.

ddn0 committed:
build: Use ad-hoc text file to carry build arguments

The ability of CMake to escape quotes in generator variables is
rudimentary. So instead of writing well-formed JSON, output simple
key=value lines.

arthurp commented:
Uuugh. I was expecting this was the way out. CMake is such a nightmare. I have no better solution.

arthurp commented on python/katana_setup.py:
```diff
@@ -184,27 +183,37 @@ def check_cython_module(name, cython_code, python_code="", extension_options=Non
+        idx = line.index("=")
+        key = line[:idx]
+        value = line[idx + 1 :]
+        result[key] = value
```
I think you can just write `key, value = line.strip().split("=", maxsplit=1)` or something similar and avoid all the indexing mess.

arthurp reviewed (approved):
That string indexing could use some love as commented below, but otherwise LGTM.
